sw/device/lib/runtime/pmp.h

```diff
@@ -0,0 +1,312 @@
+   * "Address Matching".
+   */
+  kPmpRegionConfigureNa4Unavailable,
+} pmp_region_configure_na4_result_t;
+
+/**
+ * Configures PMP address matching to Naturally aligned four-byte (NA4).
+ *
+ * When pmpcfgN is set to NA4 then the protected region range is formed from
+ * pmpaddrN and extends to the next 4 bytes.
+ *
+ * Please see the PMP specification reference and general information at the
+ * top of this header file.
+ *
+ * @param region PMP region to configure and set the address for.
+ * @param config PMP region configuration information.
+ * @param address PMP address to be set.
+ * @return `pmp_region_configure_na4_result_t`.
+ */
+PMP_WARN_UNUSED_RESULT
+pmp_region_configure_na4_result_t pmp_region_configure_na4(
+    pmp_region_index_t region, const pmp_region_config_t *config,
+    pmp_region_address_t address);
+
+/**
+ * PMP region access and address NAPOT configuration result.
+ */
+typedef enum pmp_region_configure_napot_result {
+  kPmpRegionConfigureNapotOk = kPmpRegionConfigureOk,
+  kPmpRegionConfigureNapotError = kPmpRegionConfigureError,
+  kPmpRegionConfigureNapotBadArg = kPmpRegionConfigureBadArg,
+  kPmpRegionConfigureNapotBadRegion = kPmpRegionConfigureBadRegion,
+  kPmpRegionConfigureNapotBadAddress = kPmpRegionConfigureBadAddress,
+  kPmpRegionConfigureNapotWarlError = kPmpRegionConfigureWarlError,
+  /**
+   * NAPOT size must be at least 8bytes and Power Of Two. NAPOT address
+   * must be aligned to the size. Please see:
+   * "Volume II: RISC-V Privileged Architectures V20190608-Priv-MSU-Ratified",
+   * "3.6 Physical Memory Protection", "Table 3.10" and section
+   * "Address Matching".
+   */
+  kPmpRegionConfigureNapotBadSize,
+} pmp_region_configure_napot_result_t;
+
+/**
+ * Configures PMP address matching to Naturally aligned power-of-two (NAPOT).
+ *
+ * Size ≥ 8 bytes. When pmpcfgN is set to NAPOT then the protected region range
+ * is formed from a single encoded pmpaddrN address. NAPOT ranges make use of
+ * the low-order bits of the associated address register to encode the size of
+ * the range, please see the table "3.10" of the "Address Matching" section of
+ * the specification mentioned above.
+ *
+ * Example: NAPOT encoded address of yyyy011 describes the
+ * yyyy00000 .. yyyy11111 range. The index of the first "0", determines the
+ * range size ( 2^(3 + index) ).
+ *
+ * Please see the PMP specification reference and general information at the
+ * top of this header file.
+ *
+ * @param region PMP region to configure and set the address for.
+ * @param config PMP region configuration information.
+ * @param address PMP address to be set (must be aligned to `size`).
+ * @param size PMP protected address range size (must be power-of-two).
+ * @return `pmp_region_configure_napot_result_t`.
+ */
+PMP_WARN_UNUSED_RESULT
+pmp_region_configure_napot_result_t pmp_region_configure_napot(
+    pmp_region_index_t region, const pmp_region_config_t *config,
```

Comment:
To be consistent with the DIF APIs, `config` should be taken by value, not by pointer.

Reply:
Will do.

Reply:
Done.